### ISSUE TITLE
confidence scores differ between the yolov5su model trained customly in yolov8 and the yolov5s model run in yolov5.

### ISSUE THREAD

**Issue by muhaha211** [ultralytics/ultralytics#8278]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I trained the yolov5su model using yolov8 in the following manner. When I perform inference using predict, the average confidence score differs by about 0.8 to 1 compared to when I perform inference using the yolov5s model in yolov5. Why is that?

this is how i trained yolov5s in yolov5.
```
!python train.py --img 640 --batch 64 --epochs 300 --data /content/drive/MyDrive/cho/data.yaml --cfg ./models/yolov5s.yaml --weights yolov5s.pt --name cho_yolov5_res1

YOLOv5s summary: 157 layers, 7015519 parameters, 0 gradients, 15.8 GFLOPs
                 Class     Images  Instances          P          R      mAP50   mAP50-95: 100% 1/1 [00:02<00:00,  2.28s/it]
                   all         72        374      0.995      0.998      0.995      0.851
                 medal         72         15      0.989          1      0.995      0.808
                   cho         72        359          1      0.996      0.995      0.893
```

this is for yololv8. i used same data.yaml, same images, same labels.
for inference, it doesnt matter which methods you use (for yolov5, torch.hub or detect.py and for yolov8 model.predict or just model) 
```
!yolo train data=/content/drive/MyDrive/cho/data.yaml model=yolov5s.yaml epochs=300 batch=64 imgsz=640 patience=0
YOLOv5s summary (fused): 193 layers, 9112310 parameters, 0 gradients, 23.8 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100% 1/1 [00:00<00:00,  1.13it/s]
                   all         72        374      0.991      0.996      0.995      0.865
                 medal         72         15      0.983          1      0.995      0.838
                   cho         72        359          1      0.993      0.995      0.892
```

**Comment by glenn-jocher** [comment]
@muhaha211 hey there! 👋 It's great to see you're experimenting with different YOLO versions. The difference in confidence scores between YOLOv5su trained in YOLOv8 and YOLOv5s in YOLOv5 could be due to several factors, such as differences in model architecture, training procedures, or even slight variations in the underlying codebase that handles the prediction calculations.

To ensure consistency, you might want to check that both models are using the same version of the pre-trained weights, the same hyperparameters during training, and that the inference settings (like confidence thresholds and non-maximum suppression) are aligned.

If you're still noticing discrepancies, consider running a few more tests with different images or diving deeper into the training logs to compare the loss curves and other metrics.

Here's a quick example of how to set the confidence threshold for predictions in YOLOv8:

```python
from ultralytics import YOLO

# Load your custom trained model
model = YOLO('path/to/your/custom_yolov8_model.pt')

# Set the confidence threshold for predictions
results = model.predict('path/to/your/image.jpg', conf=0.25)
```

Feel free to share any additional details or observations, and we can look into this together! 😊

**Comment by glenn-jocher** [comment]
@muhaha211 hi there! 👋 Differences in confidence scores between models can arise due to various factors, including differences in model architecture, training hyperparameters, or even slight variations in the training data preprocessing. 

In your case, since you're using the same dataset and training configuration, it's possible that the underlying changes between YOLOv5 and YOLOv8, such as updates in the model architecture or the training process, could lead to these variations in confidence scores.

To ensure consistency, you might want to check that all settings are aligned between the two versions, including any data augmentation techniques and post-processing steps like Non-Max Suppression (NMS) thresholds. If everything is consistent, the observed differences might simply be due to improvements or changes in the newer YOLOv8 version.

If you'd like to align the confidence scores more closely, you could consider fine-tuning the confidence threshold during inference or adjusting the NMS threshold. Here's a quick example of how to adjust the confidence threshold during inference with YOLOv8:

```python
from ultralytics import YOLO

# Load your custom trained model
model = YOLO('path/to/your/custom_model.pt')

# Set a custom confidence threshold for inference
results = model.predict('path/to/your/image.jpg', conf=0.25)  # Set your desired confidence threshold
```

Feel free to share any additional details or observations, and we can dive deeper into this! 😊
